hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/AsyncProtobufLogWriter.java

```diff
@@ -205,9 +205,11 @@ private long writeWALMetadata(Consumer<CompletableFuture<Long>> action) throws I
       InterruptedIOException ioe = new InterruptedIOException();
       ioe.initCause(e);
       throw ioe;
-    } catch (ExecutionException | TimeoutException e) {
+    } catch (ExecutionException e) {
       Throwables.propagateIfPossible(e.getCause(), IOException.class);
       throw new RuntimeException(e.getCause());
+    } catch (TimeoutException e) {
+      throw new RuntimeException(e);
```

Comment:
Better change this to TimeoutIOException?

Reply:
Good idea!